packages/authentication/src/providers/google.ts

```diff
@@ -0,0 +1,306 @@
+import {
+  AbstractAuthenticationModuleProvider,
+  AuthenticationResponse,
+} from "@medusajs/types"
+import { AuthProviderService, AuthUserService } from "@services"
+import jwt, { JwtPayload } from "jsonwebtoken"
+import url, { UrlWithParsedQuery } from "url"
+
+import { AuthProvider } from "@models"
+import { MedusaError } from "@medusajs/utils"
+import { OAuth2 } from "oauth"
+import { TreeLevelColumn } from "typeorm"
+
+type InjectedDependencies = {
+  authUserService: AuthUserService
+  authProviderService: AuthProviderService
+}
+
+type GoogleProviderConfig = {
+  callbackURL: string
+  clientID: string
+  clientSecret: string
+}
+
+type AuthenticationInput = {
+  connection: { encrypted: boolean }
+  url: string
+  headers:{ host: string }
+  query: Record<string, string>
+  body: Record<string, string>
+}
+
+class GoogleProvider extends AbstractAuthenticationModuleProvider {
+  public static PROVIDER = "google"
+  public static DISPLAY_NAME = "Google Authentication"
+
+  // TODO: abstract
```

Comment:
What does this mean?

Reply:
later on we will need to createa a OAuth2Service to encapsulate the common logic of the OAuth 2 mechanism and only required to implement the custom logic. So everything to be abstract will probably be reviewed to move there at some point

Reply:
it will simplify the creation of OAuth2 base providers and also help users to create custom providers without reimplementing the will everytime and be sure it is handled like we want